src/security/builtin_plugins/authentication/src/auth_utils.c

```diff
@@ -0,0 +1,1530 @@
+        result = DDS_SECURITY_VALIDATION_FAILED;
+        DDS_Security_Exception_set(ex, DDS_AUTH_PLUGIN_CONTEXT, DDS_SECURITY_ERR_UNDEFINED_CODE, (int)result, "Failed to allocate EVP key: %s", msg);
+        ddsrt_free(msg);
+        goto fail_alloc_pkey;
+    }
+
+    if (EVP_PKEY_set1_EC_KEY(*pkey, eckey) != 1) {
+        char *msg = get_openssl_error_message();
+        result = DDS_SECURITY_VALIDATION_FAILED;
+        DDS_Security_Exception_set(ex, DDS_AUTH_PLUGIN_CONTEXT, DDS_SECURITY_ERR_UNDEFINED_CODE, (int)result, "Failed to set EVP key to EC public key: %s", msg);
+        ddsrt_free(msg);
+        goto fail_pkey_set_eckey;
+    }
+
+    EC_KEY_free(eckey);
+    EC_POINT_free(point);
+    EC_GROUP_free(group);
+
+    return result;
+
+fail_pkey_set_eckey:
+    EVP_PKEY_free(*pkey);
+fail_alloc_pkey:
+fail_eckey_set_pubkey:
+fail_eckey_set_group:
+    EC_KEY_free(eckey);
+fail_alloc_eckey:
+fail_oct2point:
+    EC_POINT_free(point);
+fail_alloc_point:
+    EC_GROUP_free(group);
+fail_alloc_group:
+    return result;
+}
+
+DDS_Security_ValidationResult_t
+dh_read_public_key_by_value(
+    EVP_PKEY **pkey,
+    AuthenticationAlgoKind_t algo,
+    const unsigned char *keystr,
+		uint32_t size,
+    DDS_Security_SecurityException *ex)
+{
+    DDS_Security_ValidationResult_t result = DDS_SECURITY_VALIDATION_OK;
+
+    assert(pkey);
+    assert(keystr);
+
+    switch (algo) {
+    case AUTHENTICATION_ALGO_KIND_RSA_2048:
+        result = dh_oct_to_public_key_modp(pkey, keystr, size, ex);
+        break;
+    case AUTHENTICATION_ALGO_KIND_EC_PRIME256V1:
+        result = dh_oct_to_public_key_ecdh(pkey, keystr, size, ex);
+        break;
+    default:
+        assert(0);
+        result = DDS_SECURITY_VALIDATION_FAILED;
+        DDS_Security_Exception_set(ex, DDS_AUTH_PLUGIN_CONTEXT, DDS_SECURITY_ERR_UNDEFINED_CODE, (int)result, "Invalid key algorithm specified");
+        break;
+    }
+
+    return result;
+}
+
+char *
+string_from_data(
+    const unsigned char *data,
+    uint32_t size)
+{
+    char *str = NULL;
+
+    if (size > 0 && data) {
+        str = ddsrt_malloc(size+1);
+        memcpy(str, data, size);
+        str[size] = '\0';
+    }
+
+    return str;
+}
+
+void
+clean_ca_list( X509Seq *ca_list)
+{
+    unsigned i;
+    if( ca_list->_buffer != NULL && ca_list->_length > 0){
+        for (i = 0; i < ca_list->_length; ++i) {
+            X509_free(ca_list->_buffer[i]);
+        }
+        ddsrt_free ( ca_list->_buffer );
+    }
+    ca_list->_buffer = NULL;
+    ca_list->_length = 0;
+}
+
+DDS_Security_ValidationResult_t
+get_trusted_ca_list ( const char* trusted_ca_dir,
+                X509Seq *ca_list,
+                DDS_Security_SecurityException *ex){
+
+
+    DDS_Security_ValidationResult_t loading_result = DDS_SECURITY_VALIDATION_OK;
+    DDSRT_UNUSED_ARG( ca_list );
+    DDSRT_UNUSED_ARG( trusted_ca_dir );
+    DDSRT_UNUSED_ARG( ex );
+/* TODO: Trusted CA directory tracing function should be ported */
+#if( 0 )
```

Comment:
Let get_trusted_ca_list() return an error (not_supported?) as long as directory handling is not implemented nor used.

Reply:
I do not want to add a new return code enum since 
1. We're calling this function now
2. we're planning to implement trusted_ca_list before merging to master